test/parallel/test-http-path-contains-unicode.js

```diff
@@ -0,0 +1,64 @@
+// Copyright Joyent, Inc. and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+'use strict';
+require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const expected = '/café🐶';
+let result = '';
+
+assert.strictEqual(
+ expected,
+ '/caf\u{e9}\u{1f436}',
+ 'Sanity check that string literal produced the expected string'
+);
+
+const server = http.Server(function(req, res) {
+  result = req.url;
+  req.on('data', function() {
+  }).on('end', function() {
+    server.close();
+    res.writeHead(200);
+    res.end('hello world\n');
+  });
+
+});
+
+server.listen(0, function() {
+  http.request({
+    port: this.address().port,
+    path: expected,
+    method: 'GET'
+  }, function(res) {
+    console.log(res.statusCode);
+    res.resume();
+  }).on('error', function(e) {
+    console.log(e.message);
+    process.exit(1);
+  }).end();
+});
+
+process.on('exit', function() {
```

Comment:
If using `common.mustCall()` and inline asserts, there's no need for an exit handler.